app/Adapters/ViewiCodeIgniterAdapter.php

```diff
@@ -21,18 +21,7 @@ public function __construct(CodeIgniter $app)
 
     public function register($method, $url, $component, $defaults): void
     {
-        /** @var RouteCollection $routes */
-        $routes = Services::routes();
-
-        // replace route params `{name}` with placeholders.
-        // @TODO now supports only `{name}`.
-        $url = preg_replace('/{\w+?}/', '(:segment)', $url);
-
-        $routes->{$method}($url, static function (...$params) use ($component) {
-            $controller = new ViewiCodeIgniterComponent($component);
-
-            return $controller->index($params);
-        }, ['as' => $component]);
+        // skip, we use 404 handler to catch all Viewi routes. No need to convert route syntax.
```

Comment:
There are Pros and Cons to this way.
Pros are that there is no need to convert the syntax of the route.
Cons are that CI4 [controller filters](https://codeigniter4.github.io/CodeIgniter4/incoming/filters.html) cannot be used.

Reply:
Got your point. Btw, Viewi has middleware/guard support
https://viewi.net/docs/guards
Theoretically, it supposed to cover the filter case. But I will try and make the syntax converter, got an idea.

Reply:
@kenjis done, please check now